### DISCUSSION TITLE
Does ForecasterAutoreg employ parallel programming by default?

### DISCUSSION THREAD

**Discussion by SeyyedMahdiNoori** [JoaquinAmatRodrigo/skforecast#453] [Q&A] [answered]
I am using the ForecasterAutoreg function from the skforecast.ForecasterAutoreg for time series forecasting. It appears that this function inherently employs parallel programming to enhance computation speed. However, this presents an issue for me when executing my code on a server. Is there a way to disable this feature or restrict the number of concurrent processes?

**Comment by JoaquinAmatRodrigo** [comment, marked as answer]
Hello @SeyyedMahdiNoori,
Thank you for choosing skforecast!

While skforecast itself does not offer parallelization, there are several regressors that do. It's possible that the regressor you are utilizing in the forecaster has parallelization enabled by default. Could you please let me know which regressor you are using?

Fortunately, most implementations provide the option to specify the number of cores for parallel processing. For instance, in sklearn, you can set the parameter n_jobs=1, while in LightGBM, you can use num_threads=1.

I hope this information helps.

**Reply by SeyyedMahdiNoori** [reply]
Hi @JoaquinAmatRodrigo 

I appreciate your time.

I am studying different regressor models, ranging from linear models in the sklearn library to XGBRegressor in the xgboost library. One interesting observation I made is that these regressors utilise all available cores by default. However, I wanted to maintain consistency in my codebase when working with different regressors. As a result, I started using the threadpool_limits module from the threadpoolctl library to restrict the core usage. Something like:

with threadpool_limits(limits=1, user_api='blas'):
	rest of my code
